### ISSUE TITLE
Proxy local variables work in development, but not during build (v4)

### ISSUE THREAD

**Issue by wmakley** [middleman/middleman#1766]
The site: https://github.com/tortus/middleman-test
```middleman server``` works; ```middleman build``` does not.

__source/posts/template.html__ throws the following exception ONLY during build, not during testing:

```
undefined local variable or method `post' for #<#<Class:0x007fa2ecb82a40>:0x007fa2f2809228>
~/middleman-test/source/posts/template.html.slim:3:in `block in singleton class'
```

It is as though all locals set in the proxy are not available. I tried switching back to ERB templates, same issue. My proxy statement:

```ruby
data.blog.post.each do |id, post|
  proxy "/posts/#{post.slug}.html", "/posts/template.html", locals: { post: post, meta: {title: post.title} }, ignore: true
end
```

Since other sites work I assume I am doing something wrong, but no idea so far. It seems like the "method_missing" magic that makes the locals hash work is busted during build.

**Comment by tdreyno** [comment]
Can you test against `master` if you haven't?

**Comment by tdreyno** [comment]
Whoops, never mind, you posted the repo :) I'll investigate

**Comment by tdreyno** [comment]
Yup, fixed on master. Looked a lot like a similar bug against `ignore`. v4.1.0 coming as soon as I fix 1 bug in middleman-blog

**Comment by mgambella** [comment]
I am using Middleman 4.1.8 and ruby 2.2.5. And this is happening again: proxy local variables are working in development ( middleman server ) but NOT in build ( middleman build ) where I get instead "undefined local variable or method "

**Comment by tdreyno** [comment]
Can you open a new ticket and submit your `Gemfile.lock` and `config.rb`. Also, can you try Ruby 2.3.1?

**Comment by mgambella** [comment]
hi, Middleman code is working correctly, the error ,instead, was in my own code, so I closed this ticket as soon I discovered this. I think there is no need to reopen another ticket for this issue.

**Comment by Sinetheta** [comment]
I'm having this problem in a gem I tried fixing up middleman-blog-authors, in the "author page" proxy resource, locals are present in development but not during build. However in my project I can clearly see that `proxy` the helper, does indeed work as intended during for dev and build.

 https://github.com/Sinetheta/middleman-blog-authors/blob/master/lib/author_pages.rb#L53

The gem uses `Middleman::Sitemap::ProxyResource` directly, but I'm not sure if that's the problem, and I can't seem to get a `pry` to work inside of template during build to see what's going wrong.

I understand that this gem I've resurected may be old and awful, but I'd appreciate any hints as to what could cause a problem like this. I get this same problem on any version of middleman.
